lib/queue/worker.py

```diff
@@ -48,13 +51,68 @@ def safely_respond(self, model: Model) -> List[schemas.Message]:
         Return responses if no failure.
         """
         messages_with_queues = self.receive_messages(model.BATCH_SIZE)
-        responses = []
-        if messages_with_queues:
-            logger.debug(f"About to respond to: ({messages_with_queues})")
-            try:
-                responses = model.respond([schemas.Message(**{**json.loads(message.body), **{"model_name": model.model_name}}) for message, queue in messages_with_queues])
-            except Exception as e:
-                logger.error(e)
-            self.delete_messages(messages_with_queues)
+        if not messages_with_queues:
+            return []
+        messages = self.extract_messages(messages_with_queues, model)
+        responses = self.execute_with_timeout(model.respond, messages, timeout_seconds=TIMEOUT)
+        self.delete_processed_messages(messages_with_queues)
```

Comment:
I think we should we check if the item was successfully processed and if not maybe retry, or send to 'dead letter' queue so that it isn't lost?

Reply:
Agreed!